### PR TITLE
actually update deduct_block when paying rent

### PR DESCRIPTION
**Pull request by thiolliere**
hmm how do we forget to check this.

### CODE REVIEW

**Comment by thiolliere** [comment]
I'll try to make a better deduct block test

**Comment by thiolliere** [comment]
thanks cheme for finding that though

File: srml/contract/src/tests.rs [hunk 1]
```diff
@@ -718,7 +718,23 @@ fn deduct_blocks() {
 				* 4; // blocks to rent
 			let bob_contract = super::ContractInfoOf::<Test>::get(BOB).unwrap().get_alive().unwrap();
 			assert_eq!(bob_contract.rent_allowance, 1_000 - rent);
+			assert_eq!(bob_contract.deduct_block, 5);
 			assert_eq!(Balances::free_balance(BOB), 30_000 - rent);
+
+			// Advance 7 blocks more
```
Comment (pepyakin) [inline srml/contract/src/tests.rs]:
```suggestion
			// Advance 12 blocks more
```

File: srml/contract/src/tests.rs [hunk 2]
```diff
@@ -720,6 +720,22 @@ fn deduct_blocks() {
 			assert_eq!(bob_contract.rent_allowance, 1_000 - rent);
 			assert_eq!(bob_contract.deduct_block, 5);
 			assert_eq!(Balances::free_balance(BOB), 30_000 - rent);
+
+			// Advance 7 blocks more
+			System::initialize(&12, &[0u8; 32].into(), &[0u8; 32].into());
+
+			// Trigger rent through call
+			assert_ok!(Contract::call(Origin::signed(ALICE), BOB, 0, 100_000, call::null()));
```
Comment (cheme) [inline srml/contract/src/tests.rs]: can we also test a second call at this block (I won't asker recursive contract call in transaction as I am really not sure it is doable with the test backend).

File: srml/contract/src/tests.rs (hunk 1 above)
Comment (thiolliere) [reply srml/contract/src/tests.rs]: well either advance to block 12 or advance 7 blocks more. But not 12 blocks more

File: srml/contract/src/tests.rs (hunk 2 above)
Comment (thiolliere) [reply srml/contract/src/tests.rs]: the same call twice ?

Comment (cheme) [reply srml/contract/src/tests.rs]: yes just to ensure rent is not payed twice when two call are done on same accountdb

Comment (thiolliere) [reply srml/contract/src/tests.rs]: yes ok done

**Pull request merged**